sentry/src/main/java/io/sentry/hints/Hints.java

```diff
@@ -9,21 +13,51 @@ public final class Hints {
 
   private final @NotNull Map<String, Object> internalStorage = new HashMap<String, Object>();
 
+  public static @NotNull Hints withAttachment(@Nullable Attachment attachment) {
+    @NotNull final Hints hints = new Hints();
+    hints.getAttachmentContainer().add(attachment);
+    return hints;
+  }
+
+  public static @NotNull Hints withAttachments(@Nullable List<Attachment> attachments) {
+    @NotNull final Hints hints = new Hints();
+    hints.getAttachmentContainer().addAll(attachments);
+    return hints;
+  }
+
   public void set(@NotNull String hintType, @Nullable Object hint) {
     internalStorage.put(hintType, hint);
   }
 
-  public @Nullable Object get(@NotNull String hintType) {
-    return internalStorage.get(hintType);
+  public @Nullable Object get(@NotNull String hintName) {
+    return internalStorage.get(hintName);
   }
 
-  // TODO maybe not public
-  public void remove(@NotNull String hintType) {
-    internalStorage.remove(hintType);
+  @SuppressWarnings("unchecked")
+  public <T extends Object> @Nullable T getAs(@NotNull String hintName, @NotNull Class<T> clazz) {
+    Object hintValue = internalStorage.get(hintName);
+
+    if (clazz.isInstance(hintValue)) {
```

Comment:
@marandaneto @romtsn this is where I had trouble with Kotlin/Java interop. When I pass in e.g. `Long` it returns `null` because the `clazz.isInstance` check fails. There's a test for this that may help in explaining the issue.

Reply:
Is it OK to document this as a known limitation for now and come back later?

Reply:
I think it's risky to document this as a known limitation. On Android, almost everybody uses Kotlin. In the worst case scenario, we maybe need to change the API to make it work for Kotlin.

Reply:
It's either document or fix imho. I'd need to do more investigation on how to fix this. Maybe @marandaneto or @romtsn know how to deal with this?